crates/cubecl-reduce/src/test.rs

```diff
@@ -33,212 +35,392 @@ macro_rules! testgen_reduce {
+            .into_iter()
+            .map(|sum| sum / F::new(self.shape[self.reduce_dim as usize] as f32))
+            .collect()
+    }
+
+    fn cpu_argmax_dim<F: Float>(&self, values: &[F]) -> Vec<u32> {
+        let mut expected = vec![(F::MIN, 0_u32); self.num_output_values()];
+        for input_index in 0..values.len() {
+            let output_index = self.to_output_index(input_index);
+            let (best, _) = expected[output_index];
+            let candidate = values[input_index];
+            if candidate > best {
+                let coordinate = self.to_input_coordinate(input_index);
+                expected[output_index] = (candidate, coordinate[self.reduce_dim as usize] as u32);
+            }
+        }
+        expected.into_iter().map(|(_, i)| i).collect()
+    }
+
+    fn cpu_argmin_dim<F: Float>(&self, values: &[F]) -> Vec<u32> {
+        let mut expected = vec![(F::MAX, 0_u32); self.num_output_values()];
+        for input_index in 0..values.len() {
+            let output_index = self.to_output_index(input_index);
+            let (best, _) = expected[output_index];
+            let candidate = values[input_index];
+            if candidate < best {
+                let coordinate = self.to_input_coordinate(input_index);
+                expected[output_index] = (candidate, coordinate[self.reduce_dim as usize] as u32);
+            }
+        }
+        expected.into_iter().map(|(_, i)| i).collect()
+    }
+
+    fn num_output_values(&self) -> usize {
+        self.line_size as usize * self.shape.iter().product::<usize>()
+            / self.shape[self.reduce_dim as usize]
+    }
+
+    fn to_output_index(&self, input_index: usize) -> usize {
+        let line_size = self.line_size as usize;
+        let mut coordinate = self.to_input_coordinate(input_index / line_size);
+        coordinate[self.reduce_dim as usize] = 0;
+        self.from_output_coordinate(coordinate) * line_size + input_index % line_size
+    }
+
+    fn to_input_coordinate(&self, index: usize) -> Vec<usize> {
+        self.stride
+            .iter()
+            .zip(self.shape.iter())
+            .map(|(stride, shape)| (index / stride) % shape)
+            .collect()
+    }
+
+    fn from_output_coordinate(&self, coordinate: Vec<usize>) -> usize {
+        coordinate
+            .into_iter()
+            .zip(self.output_stride().iter())
+            .map(|(c, s)| c * s)
+            .sum()
+    }
+
+    fn output_stride(&self) -> Vec<usize> {
+        let dim_stride = self.stride[self.reduce_dim as usize];
+        let dim_shape = self.shape[self.reduce_dim as usize];
+        self.stride
+            .iter()
+            .map(|s| match s.cmp(&dim_stride) {
+                std::cmp::Ordering::Equal => 1,
+                std::cmp::Ordering::Greater => s / dim_shape,
+                std::cmp::Ordering::Less => *s,
+            })
+            .collect()
+    }
+
+    fn random_input_values<F: Float>(&self) -> Vec<F> {
```

Comment:
Quite a duplication from matmul test_utils. This would be a good case for cubecl-std, under a test cfg. Also, the matmul one has a seed as argument which can prevent bug proneness when we have several inputs. I think we should have an Option<u32> and use 123456789 when None. Anyway, not a blocker for this PR.

Reply:
I can create an issue?

Reply:
Yeah, maybe an issue about creating cubecl-std with common stuff between reduce and matmul.